### DISCUSSION TITLE
Add new host: SSL sertificate problem

### DISCUSSION THREAD

**Discussion by headlonger** [gravitational/teleport#20807] [Q&A] [answered]
Hi
I installed Teleport (for the first time) open source server v11.2.3. OS Debian 11.
**I use self-signed SSL cert.**
If I open https://teleport.my.domain:3080, then everything is fine. The connection is secure and the SSL certificate is valid. The web menu opens, authorization is successful. I can add users, roles.
I open the "Servers" menu, click add server. I copy the code and execute it on the client host to be added - `sudo bash -c "$(curl -fsSL https://teleport.my.domain:3080/scripts/<TOKEN>/install-node.sh)"` . Аnd i get the error - `An error has occurred. The token may be expired or invalid. Please check log for further details.`
I tried to generate new links with new tokens, the error is always the same.
CURL from the client host https://teleport.my.domain:3080 answer - `<a href="/web">Found</a>`.

I found information that the addition occurs through the authentication server on port 8025 (correct me if i'm wrong).
When I open https://teleport.my.domain:3025, the browser shows that the SSL certificate is not valid. In this case, the certificate was apparently generated by the Teleport server itself.
```
Issued to
Common Name (CN)	1d34567a-b125-412f-9127-a12345678c0c.server-name
Organization (O)	Admin-user-name
Organizational Unit (OU)	<Not Part Of Certificate>
Issued by
Common Name (CN)	server-name
Organization (O)	server-name
Organizational Unit (OU)	<Not Part Of Certificate>
```
Execute code on the client host to be added - `sudo bash -c "$(curl -fsSL https://teleport.my.domain:3025/scripts/<TOKEN>/install-node.sh)"` . Аnd i get the error - `(60) SSL certificate problem: unable to get local issuer certificate`
CURL from the client host https://teleport.my.domain:3025 - `(60) SSL certificate problem: unable to get local issuer certificate.`

I added the root certificate to the host client. It doesn't give anything

It seems to me that the matter is in the certificate that is generated for the auth server.
How to create or specify ssl cert correctly?
Or what is my mistake or not mine?

Thanks in advance

**Comment by webvictim** [comment, marked as answer]
You can find the actual underlying error which is causing the `An error has occurred. The token may be expired or invalid. Please check log for further details.` in the Teleport process logs. 

Based on previous experience, however, the error most often occurs because you need to set the public address for your Teleport cluster in your `/etc/teleport.yaml` config file:

```yaml
proxy_service:
  public_addr: teleport.my.domain:3080
```

Restart Teleport once you've changed this and try to run the "Add Server" wizard again. It should work this time.

FYI for future reference, using port 3025 will not work here - it needs to be the port that the Teleport proxy is running on.

**Reply by headlonger** [reply]
thanks a lot
